### PR TITLE
Change fuzzbench_qemu fuzzer

### PR DESCRIPTION
**Pull request by tokatoka**
- don't test libpng
- instead run the same harness as fuzzbench to see if it find objectives or not
- added these too to self-hosted ci

but it doesn't work as I said in discord.
this is probably due to signal handling is broken in Qemu?

also fuzzbench_fork_qemu doesn't seem to detect any new edges. can you check that too when you are back? @rmalmain

### CODE REVIEW

**Comment by tokatoka** [comment]
also we should add these guys to the test on skynet..

**Comment by tokatoka** [comment]
I forgot to say this.
but fuzzbench_fork_qemu() doesn't seem to be correctly instrumented @rmalmain

**Comment by rmalmain** [comment]
yes i realized with the ci, i am still debugging this one.
fuzzbench_qemu test works correctly locally at least now.

**Comment by tokatoka** [comment]
ok ok Rust toolchanins are broken as always!! 👎 😠

**Comment by tokatoka** [comment]
qemu fuzzer still reporting errors

**Comment by rmalmain** [comment]
yes it's normal, i also fixed the fork executor harness' signature

**Comment by tokatoka** [comment]
depends on #2496

**Comment by rmalmain** [comment]
few things:
- renaming of `EDGES_MAP_SIZE_IN_USE` into `EDGES_MAP_DEFAULT_SIZE` and `EDGES_MAP_SIZE_MAX` into `EDGES_MAP_ALLOCATED_SIZE`.
- `EdgeCoverageModule` stuff can only be built with a reference to observer through the builder. This is a first step to make the module work with different kinds of `MapObserver`
- integration of `VariableLengthMapObserver` and `ConstantLengthMapObserver` traits to be able to differentiate both map styles at compile time. for now i only implemented `VariableLengthMapObserver` for testing.

**Comment by rmalmain** [comment]
what you think of the changes @tokatoka ? does smth look bad?

**Comment by tokatoka** [comment]
it's good

File: libafl/src/executors/inprocess/inner.rs [hunk 1]
```diff
@@ -50,9 +45,7 @@ where
 
 impl<HT, OT, S> Debug for GenericInProcessExecutorInner<HT, OT, S>
 where
-    HT: ExecutorHooksTuple<S>,
```
Comment (tokatoka) [inline libafl/src/executors/inprocess/inner.rs:53]: why you deleted these?

Comment (rmalmain) [reply libafl/src/executors/inprocess/inner.rs:53]: These are just useless bounds right?

Comment (tokatoka) [reply libafl/src/executors/inprocess/inner.rs:53]: no it doesn't build now.

Comment (rmalmain) [reply libafl/src/executors/inprocess/inner.rs:53]: ah i think i have one change pending i forgot to push

File: libafl/src/observers/map/variable_map.rs [hunk 2]
```diff
@@ -268,7 +266,7 @@ where
     }
 }
 
-impl<'a, T> Deref for VariableMapObserver<'a, T>
```
Comment (tokatoka) [inline libafl/src/observers/map/variable_map.rs:271]: again, this is necessary

Comment (tokatoka) [reply libafl/src/observers/map/variable_map.rs:271]: don't delete deref

Comment (tokatoka) [reply libafl/src/observers/map/variable_map.rs:271]: nvm

**Comment by tokatoka** [comment]
@rmalmain 
there're python errors to fix

**Comment by tokatoka** [comment]
also can you fix the fuzzer examples?

**Comment by rmalmain** [comment]
TODO: Implement EdgeModule with  ConstantLengthMapObserver variant

**Pull request merged**